docs/api.md

```diff
@@ -6,6 +6,171 @@ challenges presented by JavaScript and web browsers.
 Please direct clarifications or observations of inaccuracy to [our issue
 tracker](https://github.com/iodide-project/iodide/issues/new).
 
+
+
+## `iodide.file`
+
+The `iodide.file` API provides convenience functions around working with files uploaded to the Iodide server in your notebook.
+
+
+
+### `iodide.file.save(data, fileName[, saveOptions])`
+
+Returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises), which, when resolved, will signal that `data` was uploaded to the server under the filename `fileName`.
+
+`data` (required) is any object or variable in the eval name space. It can be anything, really - an array of data, or a csv, or a binary blob.
+
+`filename` (required) is a string that represents the file name.
+
+The optional argument `saveOptions` has the following keys:
+
+- `overwrite` (optional, default `false`): if `true`, will overwrite whatever is at `fileName` with `data`. If `false`, will throw an error and halt the subsequent enqueued evaluation (similar to a syntax error).
+- `serializer` (optional): a function that transforms `data` to its final form.
```

Comment:
This may not be necessary, vs. just serializing `data` before passing into `iodide.file.save`

Reply:
Maybe we could not specify it for now, and add the argument later if it turns out be helpful

Reply:
Good idea.